### DISCUSSION TITLE
Does anyone have a script to split cypress tests into separate files?

### DISCUSSION THREAD

**Discussion by gajus** [cypress-io/cypress#20918] [General]
According to documentation, `it` statements are not performant. The example given in the documentation is:

```js
describe('my form', () => {
  before(() => {
    cy.visit('/users/new')
    cy.get('#first').type('johnny')
  })

  it('has validation attr', () => {
    cy.get('#first').should('have.attr', 'data-validation', 'required')
  })

  it('has active class', () => {
    cy.get('#first').should('have.class', 'active')
  })

  it('has formatted first name', () => {
    cy.get('#first').should('have.value', 'Johnny') // capitalized first letter
  })
})
```

Unfortunately, we do have many many `it` statements per test suite, and it is suspected to be the cause of [freezing tests](https://github.com/cypress-io/cypress/issues/8206#issuecomment-1086950214).

Does anyone have a script to split cypress tests into separate files?, i.e. the above would become 3 separate files:

```js
describe('my form', () => {
  before(() => {
    cy.visit('/users/new')
    cy.get('#first').type('johnny')
  })

  it('has validation attr', () => {
    cy.get('#first').should('have.attr', 'data-validation', 'required')
  })
})
```

```js
describe('my form', () => {
  before(() => {
    cy.visit('/users/new')
    cy.get('#first').type('johnny')
  })

  it('has validation attr', () => {
    cy.get('#first').should('have.attr', 'data-validation', 'required')
  })
})
```

```js
describe('my form', () => {
  before(() => {
    cy.visit('/users/new')
    cy.get('#first').type('johnny')
  })

  it('has active class', () => {
    cy.get('#first').should('have.class', 'active')
  })
})
```

and would this even alleviate the issue?

We would still like to maintain these tests as large test suites in our source control, but split them up before running tests.

**Comment by geromegrignon** [comment]
Hi @gajus,

Cypress [Best Practices](https://docs.cypress.io/guides/references/best-practices#Creating-tiny-tests-with-a-single-assertion) regarding `it` are about making longer tests to avoid atomic test suites.

> It is common for tests in Cypress to issue 30+ commands.
